lib/supavisor/tenants/user.ex

```diff
@@ -0,0 +1,48 @@
+defmodule Supavisor.Tenants.User do
+  @moduledoc false
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  @type t :: %__MODULE__{}
+
+  @primary_key {:id, :binary_id, autogenerate: true}
+  @schema_prefix "_supavisor"
+
+  schema "users" do
+    field(:db_user_alias, :string)
+    field(:db_user, :string)
+    field(:db_password, Supavisor.Encrypted.Binary, source: :db_pass_encrypted)
+    field(:is_manager, :boolean, default: false)
+    field(:mode_type, Ecto.Enum, values: [:transaction])
+    field(:pool_size, :integer)
+    belongs_to(:tenant, Supavisor.Tenants.Tenant, foreign_key: :tenant_external_id, type: :string)
+    timestamps()
+  end
+
+  @doc false
+  def changeset(user, attrs) do
+    attrs =
+      if attrs["db_user_alias"] do
+        attrs
+      else
+        Map.put(attrs, "db_user_alias", attrs["db_user"])
+      end
+
+    user
+    |> cast(attrs, [
+      :db_user_alias,
+      :db_user,
+      :db_password,
+      :pool_size,
+      :mode_type,
+      :is_manager
```

Comment:
double-checking: how important is is_manager field? Can any API user set it? Usually what we might want to do is allow setting this field only on creation but not on update. Or, allow it on update but only under certain circumstances. In those cases it's usually better not to have the field in the cast allowlist but instead set it manually using functions like `put_change` when those extra conditions are met.

Reply:
This field is not currently in use, but we plan to use it for internal operations in the future, such as obtaining the `max_connections` value. Thank you for the relevant thoughts, I'll keep them in mind 🔥